numba/tests/test_dispatcher.py

```diff
@@ -1468,6 +1468,61 @@ def execute_with_input():
         err = execute_with_input()
         self.assertIn("cache hits = 1", err.strip())
 
+    def test_ipykernel(self):
+        # Test caching in an IPython session using ipykernel
+        base_cmd = [sys.executable, '-m', 'IPython']
+        base_cmd += ['--quiet', '--quick', '--no-banner', '--colors=NoColor']
+        try:
+            ver = subprocess.check_output(base_cmd + ['--version'])
+        except subprocess.CalledProcessError as e:
+            self.skipTest("ipython not available: return code %d"
+                            % e.returncode)
+        ver = ver.strip().decode()
+        print("ipython version:", ver)
+        # Create test input
+        from ipykernel import compiler
```

Comment:
Test failing on older `ipykernel` version. Needs to skip test on old version.

Please see details at https://dev.azure.com/numba/numba/_build/results?buildId=10241&view=logs&j=a097410b-814c-5303-e822-b655d380dc70&t=4bbe663b-9287-58fe-d45f-064bba14e36f&l=377

Reply:
Also, `ipykernel` is an optional dependency. 

See https://dev.azure.com/numba/numba/_build/results?buildId=10241&view=logs&j=0b09a2a2-0f6e-53e2-0b1f-738b2a0d11e7&t=b20b5b58-0cbd-56fd-a457-b640af713e05&l=110

Reply:
Thanks, added a `pytest.importorskip("ipykernel", minversion="6.0.0")` in the test, hopefully that fixes it.